### PR TITLE
[fix](Azure) Use correct http range assignation to Azure Block Blob Client

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

The DownloadToBlobOptions' range field is one `Nullable` type which should be directly use operator= to assign one value.
Also this pr tries to print the e.what message for better debug.

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

File: be/src/io/fs/azure_obj_storage_client.cpp [hunk 1]
```diff
@@ -84,9 +85,9 @@ ObjectStorageUploadResponse AzureObjStorageClient::upload_part(const ObjectStora
         Azure::Core::IO::MemoryBodyStream memory_body(
                 reinterpret_cast<const uint8_t*>(stream.data()), stream.size());
         client.StageBlock(std::to_string(part_num), memory_body);
-    } catch (Azure::Storage::StorageException& e) {
```
Comment (gavinchou) [inline be/src/io/fs/azure_obj_storage_client.cpp:87]: Also catch std::exception

**Review by gavinchou — CHANGES_REQUESTED** [review]
Also catch std::exception

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**